src/reanimated2/hook/useAnimatedRef.ts

```diff
@@ -13,35 +13,50 @@ import { isFabric, isWeb } from '../PlatformChecker';
 
 const IS_WEB = isWeb();
 
-interface MaybeScrollableComponent extends Component {
-  getNativeScrollRef?: FlatList['getNativeScrollRef'];
-  getScrollableNode?:
-    | ScrollView['getScrollableNode']
-    | FlatList['getScrollableNode'];
-  viewConfig?: {
-    uiViewClassName?: string;
-  };
-}
-
-function getComponentOrScrollable(component: MaybeScrollableComponent) {
-  if (isFabric() && component.getNativeScrollRef) {
-    return component.getNativeScrollRef();
-  } else if (!isFabric() && component.getScrollableNode) {
-    return component.getScrollableNode();
-  }
-  return component;
-}
-
 /**
  * Lets you get a reference of a view that you can use inside a worklet.
  *
  * @returns An object with a `.current` property which contains an instance of a component.
  * @see https://docs.swmansion.com/react-native-reanimated/docs/core/useAnimatedRef
  */
-export function useAnimatedRef<
+export const useAnimatedRef: UseAnimatedRef = IS_WEB
+  ? useAnimatedRefWeb
+  : useAnimatedRefNative;
+
+function useAnimatedRefWeb<
+  TComponent extends Component
+>(): AnimatedRef<TComponent> {
+  const componentReference = useSharedValue<HTMLElement | null>(null);
+
+  const ref = useRef<AnimatedRef<TComponent>>();
+
+  if (!ref.current) {
+    const fun: AnimatedRef<TComponent> = <AnimatedRef<TComponent>>((
+      component
+    ) => {
+      // Enters when ref is set by attaching to a component.
+      if (component) {
+        componentReference.value = getComponentOrScrollable(component);
+        fun.getTag = () => getTagFromComponent(component);
```

Comment:
If I understand correctly, this code changes the function's behavior. Previously, on the web platform, the `getTag` function had this implementation:
```js
fun.getTag = () => getComponentOrScrollable(component)
``` 

Is it intentionally?